arlo-client/src/components/Audit/Setup/Participants/useJurisdictionManifest.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react'
+import { toast } from 'react-toastify'
+import { api } from '../../../utilities'
+import { IFileInfo } from '../../useJurisdictions'
+
+const loadJurisdictionManifest = async (
+  electionId: string
+): Promise<IFileInfo | null> => {
+  try {
+    return await api(`/election/${electionId}/jurisdiction/file`)
+  } catch (err) {
+    toast.error(err.message)
+    return null
+  }
+}
+
+const putJurisdictionManifestFile = async (
+  electionId: string,
+  csv: File
+): Promise<boolean> => {
+  const formData: FormData = new FormData()
+  formData.append('jurisdictions', csv, csv.name)
+  try {
+    await api(`/election/${electionId}/jurisdiction/file`, {
+      method: 'PUT',
+      body: formData,
+    })
+    return true
+  } catch (err) {
+    toast.error(err.message)
+    return false
+  }
+}
+
+const useJurisdictionManifest = (
+  electionId: string
+): [IFileInfo | null, (csv: File) => Promise<boolean>] => {
+  const [
+    jurisdictionManifest,
+    setJurisdictionManifest,
+  ] = useState<IFileInfo | null>(null)
+
+  const uploadJurisdictionManifest = async (csv: File): Promise<boolean> => {
+    // TODO poll for result of upload
+    if (await putJurisdictionManifestFile(electionId, csv)) {
+      setJurisdictionManifest(await loadJurisdictionManifest(electionId))
+      return true
+    }
+    return false
+  }
+
+  useEffect(() => {
+    ;(async () => {
+      setJurisdictionManifest(await loadJurisdictionManifest(electionId))
+    })()
+  }, [electionId])
+
+  return [jurisdictionManifest, uploadJurisdictionManifest]
+}
+
+export default useJurisdictionManifest
```

Comment:
Small suggestion to name this `useJurisdictionsFile` and reserve the word "manifest" to refer specifically to the ballot manifest